helm/botkube/templates/sslcert.yaml

```diff
@@ -1,15 +1,13 @@
-{{- if .Values.config.ssl.enabled }}
-
+{{- if and .Values.config.ssl.enabled .Values.config.ssl.cert }}
 apiVersion: v1
 kind: Secret
 metadata:
-  name: {{ include "botkube.fullname" . }}-certificate-secret
+  name: {{ include "botkube.SSLCertSecretName" . }}
   labels:
     app.kubernetes.io/name: {{ include "botkube.name" . }}
     helm.sh/chart: {{ include "botkube.chart" . }}
     app.kubernetes.io/instance: {{ .Release.Name }}
     app.kubernetes.io/managed-by: {{ .Release.Service }}
 data:
   ca-certificates.crt: {{ .Files.Get (printf "%s" .Values.config.ssl.cert) | b64enc }}
```

Comment:
The changes you made work fine 👍 However, I wasn't able to test the point no 3 from the instruction—no matter what kind of file path I provide (relative/absolute), the `ca-certificates.crt` in the secret is always empty.

What I do:

```bash
cat > /tmp/ssl.crt << ENDOFFILE
-----BEGIN CERTIFICATE-----
MIICsDCCAZgCCQCmMkVyXildtDANBgkqhkiG9w0BAQsFADAaMQswCQYDVQQGEwJV
UzELMAkGA1UEAwwCQ0EwHhcNMjIwNTMwMDY1MzM3WhcNMjMwNTMwMDY1MzM3WjAa
MQswCQYDVQQGEwJVUzELMAkGA1UEAwwCQ0EwggEiMA0GCSqGSIb3DQEBAQUAA4IB
DwAwggEKAoIBAQDBQxrdQkUeTinfAELgwyh5k0lFyVIiyFjtE/NiMhOCXj+fcRuv
klZRG5VVldUBUnH69Ai2BccMt0PvD7XFAFGWLyxpdyICdZpwKJhI6WjSN7mxHABh
4ELNgTvFKuP5wwt9j8+oBdeJqMrivToweIE8yCKdHsyWEFZ1Tej9IIJDF4SAVypn
U8lnv3lB70nhCXJ0fFcBajEUx+rTaMDnhy5nxlRZMBa3S5TlXC7XOyGm6ER9OP9o
Nhs7Jh0b0/+qvXnVQYB+k8ew7WUa+7ANWZI+RD6k4C5mORZy5A4jPbWgG5OkIV4I
QVO+BEqkVkE60jg3Ams2/IUMNwE2CVi2r3y9AgMBAAEwDQYJKoZIhvcNAQELBQAD
ggEBACTGmrwq7NDt8QXp5R/l0XYDBN/Qc9bsOszvcmGV0WTBmp/ScfDYBZD4EwYO
RsAlmw9+LJbZrsMcdMTt/kExKsD739e3jstBaA79+jNVzDIzDesrVYo6W8lMzbYi
lJ0sH2I+nzcgJxfE/K2Jvd7qitSXv5ne++ZozlNWqzWefbDWbg0oZDT2km1Exbyv
bFtZiu80R583Fowdk6UWGljtRTEB0ihOTc1p8LyUIYSHDPQlJfrDEpDisT8SQkyE
n8MdL5qa2I3iUMzmwfI0zzrCAtSmvjpar7ymqTlOwSF8kFw2CZu1U7LWAHTa2/PV
6r7KfL6ZBxZUEkKf8RKUpndEF0U=
-----END CERTIFICATE----
ENDOFFILE
```

```bash
helm template ./helm/botkube/ --namespace botkube --set config.ssl.enabled=true --set config.ssl.cert=/tmp/ssl.crt
```

And I get
```yaml
# Source: botkube/templates/sslcert.yaml
apiVersion: v1
kind: Secret
metadata:
  name: release-name-botkube-certificate-secret
  labels:
    app.kubernetes.io/name: botkube
    helm.sh/chart: botkube-v0.12.4
    app.kubernetes.io/instance: release-name
    app.kubernetes.io/managed-by: Helm
data:
  ca-certificates.crt: # <- empty value
```

And of course helm install fails, cause:
```
Error: INSTALLATION FAILED: unable to build kubernetes objects from release manifest: error validating "": error validating data: unknown object type "nil" in Secret.data.ca-certificates.crt
```

@mszostok Please take a look if you also encounter such issue. Thanks!

BTW I use Helm 3.9.0.

Reply:
Alright, it was a user error - @mszostok guided me to place this file inside Helm chart, and this works perfectly:

```
cat > ./helm/botkube/ssl.crt << ENDOFFILE
-----BEGIN CERTIFICATE-----
MIICsDCCAZgCCQCmMkVyXildtDANBgkqhkiG9w0BAQsFADAaMQswCQYDVQQGEwJV
UzELMAkGA1UEAwwCQ0EwHhcNMjIwNTMwMDY1MzM3WhcNMjMwNTMwMDY1MzM3WjAa
MQswCQYDVQQGEwJVUzELMAkGA1UEAwwCQ0EwggEiMA0GCSqGSIb3DQEBAQUAA4IB
DwAwggEKAoIBAQDBQxrdQkUeTinfAELgwyh5k0lFyVIiyFjtE/NiMhOCXj+fcRuv
klZRG5VVldUBUnH69Ai2BccMt0PvD7XFAFGWLyxpdyICdZpwKJhI6WjSN7mxHABh
4ELNgTvFKuP5wwt9j8+oBdeJqMrivToweIE8yCKdHsyWEFZ1Tej9IIJDF4SAVypn
U8lnv3lB70nhCXJ0fFcBajEUx+rTaMDnhy5nxlRZMBa3S5TlXC7XOyGm6ER9OP9o
Nhs7Jh0b0/+qvXnVQYB+k8ew7WUa+7ANWZI+RD6k4C5mORZy5A4jPbWgG5OkIV4I
QVO+BEqkVkE60jg3Ams2/IUMNwE2CVi2r3y9AgMBAAEwDQYJKoZIhvcNAQELBQAD
ggEBACTGmrwq7NDt8QXp5R/l0XYDBN/Qc9bsOszvcmGV0WTBmp/ScfDYBZD4EwYO
RsAlmw9+LJbZrsMcdMTt/kExKsD739e3jstBaA79+jNVzDIzDesrVYo6W8lMzbYi
lJ0sH2I+nzcgJxfE/K2Jvd7qitSXv5ne++ZozlNWqzWefbDWbg0oZDT2km1Exbyv
bFtZiu80R583Fowdk6UWGljtRTEB0ihOTc1p8LyUIYSHDPQlJfrDEpDisT8SQkyE
n8MdL5qa2I3iUMzmwfI0zzrCAtSmvjpar7ymqTlOwSF8kFw2CZu1U7LWAHTa2/PV
6r7KfL6ZBxZUEkKf8RKUpndEF0U=
-----END CERTIFICATE----
ENDOFFILE

helm template ./helm/botkube/ --namespace botkube --set config.ssl.enabled=true --set config.ssl.cert=ssl.crt
```

Quite counterintuitive behavior from Helm (as it doesn't say anything even when running it with `--debug` flag). Anyway, the issue is resolved 😄